e2e-tests/src/tests/external-datasets.test.ts

```diff
@@ -341,3 +344,231 @@ test.describe.serial('External Datasets', () => {
+
+    expect(result).toEqual(profile_set_result);
+  });
+
+  test("Check there is one violation when simulationDatasetId isn't provided", async ({ request }) => {
+    const violations: ConstraintViolation[] = await req.checkConstraints(request, plan_id);
+
+    expect(violations).not.toBeNull();
+    expect(violations).toBeDefined();
+    expect(violations).toHaveLength(1);
+  });
+
+  test("Check there is one violation when simulationDatasetId is provided", async ({ request }) => {
+    const violations: ConstraintViolation[] = await req.checkConstraints(request, plan_id, first_simulation_dataset_id);
+    violation = violations[0];
+
+    expect(violations).not.toBeNull();
+    expect(violations).toBeDefined();
+    expect(violations).toHaveLength(1);
+  });
+
+  test("Check the violation is the expected one", async () => {
+    expect(violation.constraintName).toEqual(constraint_name)
+    expect(violation.constraintId).toEqual(constraint_id)
+    expect(violation.associations.resourceIds).toHaveLength(1);
+    expect(violation.associations.resourceIds).toContain("/my_boolean");
+  });
+
+  test("Check violation starts and ends as expected", async () => {
+    const plan_start_unix = 1000 * time.getUnixEpochTime(plan_start_timestamp);
+    const dataset_start_unix = 1000 * time.getUnixEpochTime(dataset_start_timestamp);
+
+    expect(violation.windows[0].start).toEqual(dataset_start_unix - plan_start_unix + profile_duration);
+    expect(violation.windows[0].end).toEqual(dataset_start_unix - plan_start_unix + (2 * profile_duration));
+  });
+
+  test("Add activity to plan to make simulation out of date", async ({ request }) => {
+    const activityToInsert : ActivityInsertInput = {
+      arguments : {
+        "biteSize": 1
+      },
+      plan_id: plan_id,
+      type : "BiteBanana",
+      start_offset : "1h",
+    };
+    activity_id = await req.insertActivity(request, activityToInsert);
+
+    expect(activity_id).not.toBeNull();
+    expect(activity_id).toBeDefined();
+  });
+
+  test("Run new simulation", async ({ request }) => {
+    const resp: SimulationResponse = await awaitSimulation(request, plan_id);
+    second_simulation_dataset_id = resp.simulationDatasetId;
+
+    expect(resp.status).toEqual("complete");
+  });
+
+  test("Check there is still one violation when simulationDatasetId isn't provided", async ({ request }) => {
+    const violations: ConstraintViolation[] = await req.checkConstraints(request, plan_id);
+
+    expect(violations).not.toBeNull();
+    expect(violations).toBeDefined();
+    expect(violations).toHaveLength(1);
+  });
+
+  test("Check there are no violations when simulationDatasetId is provided", async ({ request }) => {
+    const violations: ConstraintViolation[] = await req.checkConstraints(request, plan_id, second_simulation_dataset_id);
+
+    expect(violations).not.toBeNull();
+    expect(violations).toBeDefined();
+    expect(violations).toHaveLength(0);
+  });
```

Comment:
Minor, but what if the first simId is provided after simulating a second time? I expect it will give the same results as `Check there is one violation when simulationDatasetId is provided`

Reply:
Good call, an additional test has been added for this case 👍